### PR TITLE
Make sure proxy class is reused when Enhancer#setClassLoader is not used

### PR DESCRIPTION
**Pull request by vlsi**
see #61

### CODE REVIEW

File: cglib/src/test/java/net/sf/cglib/proxy/TestEnhancer.java [hunk 1]
```diff
@@ -324,11 +324,25 @@ public int accept(Method method) {
 
         };
 
+        // Need to test both orders: "null classloader first" and "null last" just in case
+        Object source_ = enhance(eaClassFromCustomClassloader, null, callbackFilter, TEST_INTERCEPTOR, null);
         Object source = enhance(eaClassFromCustomClassloader, null, callbackFilter, TEST_INTERCEPTOR, custom);
+        assertEquals("Same proxy class is expected since Enhancer with null (default) ClassLoader should use " +
```
Comment (elrodro83) [inline cglib/src/test/java/net/sf/cglib/proxy/TestEnhancer.java]: Shouldn't it be assertSame?

File: cglib/src/main/java/net/sf/cglib/core/AbstractClassGenerator.java [hunk 2]
```diff
@@ -75,6 +75,9 @@ public Object apply(AbstractClassGenerator gen) {
         };
 
         public ClassLoaderData(ClassLoader classLoader) {
+            if (classLoader == null) {
```
Comment (raphw) [inline cglib/src/main/java/net/sf/cglib/core/AbstractClassGenerator.java]: Would that not cause a regression when enhancing a system class like `java.lang.Object`?

Comment (vlsi) [inline cglib/src/main/java/net/sf/cglib/core/AbstractClassGenerator.java]: Current cglib assumes class loader is not null. Otherwise it will fail with NPE here: https://github.com/cglib/cglib/blob/ef8fdd49bc71932486d4bd2d489414f8452f579a/cglib/src/main/java/net/sf/cglib/core/ReflectUtils.java#L409

Comment (raphw) [inline cglib/src/main/java/net/sf/cglib/core/AbstractClassGenerator.java]: True, I just digged a little and it seems like all classes are loaded by the system class loader in such cases.
